uwblueprint/community-fridge-kw
# feat: edit content pages

## Brief description. What is this change? 

### [Edit Content Pages](https://www.notion.so/uwblueprintexecs/Edit-Content-Pages-8f4b43e781da46ec883048fee97b1411)


## Implementation description. How did you make this change?
* Added routes for edit check in description and edit food rescue description pages
* Implemented a page allowing users to edit description/url of check ins or donations
  * `EditCheckInOrFoodRescue.tsx` component is used for both of these and the use case is differentiated by `isCheckInView`
  * Save changes will return to the corresponding view (check in/donation) page with the new description/url (`updateContent` endpoint used)
  * Clicking on the close button will return to the corresponding view with no updates
* Updated `FridgeFoodRescueDescription.tsx` and `FridgeCheckInDescription.tsx`
  * it now gets data from the content API instead of static data
  * allowed edit buttons to redirect users to the corresponding edit content page


## Steps to test
1. Use Postman to POST content to `localhost:5000/content/` with a body like this:
```
{
    "id": "1",
    "foodRescueDescription": "hi1",
    "foodRescueUrl": "google.org",
    "checkinDescription": "This is a daily checin",
    "checkinUrl": "google.org"
}
```
2. Head to admin view checkin/donations pages
3. Click edit button
4. Try changing the description and/or URL, press save changes. This should take you back to the view (check in or donation) page you were on and the content/url should be updated.
5. Go to step 2-3 again. Try changing the description and/or URL and press the close icon on the top right. This should take you back to the view (check in or donation) page you were on and the content/url should not be updated.

## Checklist
